crates/network/src/any/mod.rs

```diff
@@ -46,11 +68,537 @@ impl From<TxType> for AnyTxType {
     }
 }
 
+/// Memoization for deserialization of [`AnyTxEnvelope`] and [`AnyTypedTransaction`].
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+#[allow(unnameable_types)]
+pub struct DeserMemo {
+    input: OnceLock<Bytes>,
+    access_list: OnceLock<AccessList>,
+    blob_versioned_hashes: OnceLock<Vec<B256>>,
+    authorization_list: OnceLock<Vec<SignedAuthorization>>,
+}
+
+/// Transaction envelope for a catch-all network.
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(untagged)]
+#[doc(alias = "AnyTransactionEnvelope")]
+pub enum AnyTxEnvelope {
+    /// An Ethereum transaction.
+    Ethereum(TxEnvelope),
+    /// A transaction with unknown type.
+    Other {
+        /// Transaction type.
+        #[serde(rename = "type")]
+        ty: AnyTxType,
+
+        /// Transaction hash.
+        hash: B256,
+
+        /// Additional fields.
+        fields: std::collections::BTreeMap<String, serde_json::Value>,
+
+        /// Memoization for deserialization.
+        #[serde(skip, default)]
+        memo: DeserMemo,
+    },
```

Comment:
imo this should be it's own Standalone type, dealing with a lot of named fields is not that convenient